### ISSUE TITLE
Agile Activities - Projected Start Date

### ISSUE THREAD

**Issue by courtneycorwin** [ombegov/ITDB-schema#189]
Based on the published schema, it looks like projected completion date and projected total costs are now included, but not projected start date.  Is this going to be fixed?  If so, when?  Thanks!

**Comment by TsitsiL** [comment]
Yes that's correct, we included just those two fields precisely for the variance calculations. OMB still needs to make a decision on the projected start date if it should be included for agile activities or not.

**Comment by courtneycorwin** [comment]
Great, thanks for letting me know!  I've omitted it from our map files for now.

**Comment by rydougherty** [comment]
Hi Tsitsi- following up here. When does OMB anticipate to have a final decision regarding the Projected Start Date field?

**Comment by courtneycorwin** [comment]
Hi Tsitisi,

It looks like projected start date has been added back into the schema. I've updated our users, but was this announced somewhere?

Thanks!
Courtney

**Comment by bill-gehrke-hhs** [comment]
To add on to this, when can we expect a "release" version of the XSD that reflects these changes?  I would expect to see a new release of the XSD to be published before or at the same time as changes are made to the production system, and it appears the current release (v1.9.10) does not contain any projected fields for agile releases.  With projected fields now being made mandatory (when actual results are not available,) the latest release schema is now incompatible with the IT Dashboard.

Also, can we expect an update to the Capital Planning guidance?  The guidance linked from A-11 does not indicate that projected data is part of the business case submission for agile releases.

**Comment by TsitsiL** [comment]
Good afternoon,

**Response to Courtney:** Yes the projected start date was re-added for agile activities in the latest release. My apologies that this was not communicated to the vendors sooner. Moving forward, we will be sure to communicate schema changes in the beginning of sprints to give vendors enough time to update their tools. The next schema changes are to be deployed on 12/18. I am working on getting that list of changes communicated to vendors by early next week, which would give about a 2 week time frame for vendors to update their tools.

**Response to Bill:** The latest schema on GitHub was updated 4 days ago and does include the projected values for agile activities:[ITDB-schema/src/BusinessCase/Exchange/serviceRequest.xsd](https://github.com/ombegov/ITDB-schema/blob/master/src/BusinessCase/Exchange/serviceRequest.xsd)
I agree with you that the current version of the guidance does not include projected data for agile activities, which is why we initially removed these elements from our schema. But when we consulted with OMB, we were instructed to at least re-add the projected total cost and projected completion date fields because these are used in project variance scores which are mandated by GAO. We added back the projected start date to be consistent and it is NOT a required field in the schema. The behavior and validations of these fields are identical to what is implemented for non-agile activities.
OMB advised us that the guidance was at some point edited to re-include the projected values, but the edits were lost during version control. The intention, however, still remained - projected values were to be re-added for agile activities as optional fields. Which is what was implemented.

My apologies that this was not communicated to the vendors sooner. Moving forward, we will be sure to communicate schema changes in the beginning of sprints to give vendors enough time to update their tools. I understand from eCPIC that they were able to update their tool to accept the projected start date. The next schema changes are to be deployed on 12/18. I am working on getting that list of changes communicated to vendors by early next week, which would give about a 2 week time frame for vendors to update their tools in time to work with the changes.

**Comment by courtneycorwin** [comment]
Hi Tsitsi,

A couple of things:

1. The issue here isn't that we weren't able to update our tool quickly in order to accommodate the addition of the projected start date for agile activities.  In fact, the field and data was already in our tool, we just weren't sending it up to OMB because it was rejected with an error.   The issue also is not that we weren't able to update our schemas quickly once we found out the data element was added *back* into the schema - the issue is that we were reacting instead of being proactive.  

Letting us know what will be in the 12/18 schema update will be very helpful.  Again, it's not that we can't accommodate these changes quickly - we just want a chance to be able to test the update prior to issuing them to our user community before they start their monthly submission process.  In this case, we were issuing a schema update about 5 business days before the submissions were due.

2.  A missing projected start date does return an error if no actual start date is submitted - hence how we were notified of the fact that it needed to be added back into our schema.  So, saying it's optional is not totally correct.

3. I think what Bill was trying to point out is that we don't necessarily want the SRC file to be where we go looking for what's live.  Our understanding of that file is that it also includes updates that may not have been published yet, so while it's a "better" source for up-to-date changes, we don't have an obvious way of knowing what's been pushed to production and what's still pending.

TL/DR: Our issue is more with the process (which I know OMB is committed to improving).   While we have the ability to react very quickly to a majority of the changes OMB makes, communication about what changes are coming would be helpful, refraining from major changes (i.e. things that aren't priority bug fixes) so close to a deadline would be helpful, and having more insight to what's currently live vs. what's currently schedule to be pushed to production would also be helpful.

**Comment by TsitsiL** [comment]
Good afternoon Courtney,

I understand the issue here is the process of communicating schema changes well ahead before they are implemented. My apologies again that this did not happen with our latest deployment. Usually the schema changes make it into the Release Notes prior to deployment, but we found that this was not effective in the past in alerting vendors in time. So we covered schema changes during our weekly calls, which unfortunately had ended by the time this last release went through. Again, my apologies, and we are working to improve this process to give you ample time to update your tools. I intend to have the 12/18 schema changes communicated to all early next week around 12/4. The task for these changes is still in the beginning of development, and I am working with the developer to get the exact changes so we give you the correct list, on time.

I provided the schema location to Bill as I knew he went looking for it. The schema updates on GitHub are not intended to replace the communication to vendors that is supposed to happen whenever we make schema changes. I simply pointed Bill to where the schema is. Schema changes still need to be communicated to vendors separately. We did not do so for the last release, and this is our fault.

For the projected start date, although it is optional in the schema, it does become required when the actual start date is missing. So you are correct that it is conditionally required. This conditional behavior is enforced in the API. In terms of the schema itself, this element is listed as an optional field - which is what we implemented. 

For the schema changes in general, we do refrain from major changes as we know the impact this has on the vendors. As I mentioned before, we will do better and communicate them as soon as we know what they are exactly. 

For the 12/18 release, as I have mentioned, I am working to get the changes out to you around the 12/4. Right now we are in development and still uncovering work to be done, so I'd like to get the list when it has been finalized of what is going into production. The schema changes for 12/18 are intended to make the schema match the baseline rules. This is especially important as we prepare for Passback submissions in early January.

Many thanks.
